rest-api-spec/src/main/resources/rest-api-spec/test/search/140_pre_filter_search_shards.yml

```diff
@@ -180,5 +178,47 @@ setup:
   - match: { _shards.successful: 3 }
   - match: { _shards.skipped : 0 }
   - match: { _shards.failed: 0 }
-  - match: { hits.total: 0 }
+  - match: { hits.total.value: 0 }
   - length: { aggregations.idx_terms.buckets: 0 }
+
+  # check field sort is correct when skipping query phase
+  - do:
+      search:
+        # ensure that one shard can return empty response
+        max_concurrent_shard_requests: 1
+        pre_filter_shard_size: 1
+        body:
+          "size": 1
+          "track_total_hits": 1
+          "sort": [{ "created_at": "desc"}]
+
+  - match: { _shards.total: 3 }
+  - match: { _shards.successful: 3 }
+  - match: { _shards.skipped: 0 }
```

Comment:
I am wondering if we send `MatchNoneQueryBuilder()` request to a shard, or `size =0` for a request containing query part, can we say that we skipped this shard?

Reply:
I have mixed feelings about this one. It would provide a valuable feedback for users since rewriting to MatchNoneQueryBuilder means that we can avoid the search threadpool but that information should be materialized for users with fast response time too. It will also make the counter non-deterministic so I wonder how much confusion this would bring.

Reply:
I see, makes sense to keep it as it now.